src/lua/fiber.c

```diff
@@ -810,6 +810,113 @@ lbox_fiber_stall(struct lua_State *L)
 	return 0;
 }
 
+/* Increase deadline timeout of currently running fiber. */
+static int
+lbox_extend_slice(struct lua_State *L)
+{
+	if (lua_gettop(L) != 2) {
+		luaL_error(L, "fiber.extend_slice(fib_id, slice): "
+			      "bad arguments");
+	}
+	struct fiber *fiber = lbox_checkfiber(L, 1);
+	if (fiber != fiber()) {
+		luaL_error(L, "fiber is not running");
+	}
+	struct fiber_slice slice;
+	if (lua_istable(L, 2)) {
+		lua_getfield(L, 2, "warn");
+		slice.warn = luaL_checknumber(L, -1);
+		lua_getfield(L, 2, "err");
+		slice.err = luaL_checknumber(L, -1);
+		lua_pop(L, 2);
+	} else if (lua_isnumber(L, 2)) {
+		slice.warn = INFINITY;
+		slice.err = lua_tonumber(L, 2);
+	} else {
+		luaL_error(L, "slice must be a table or a number");
+		return 0;
+	}
+	if (slice.warn <= 0 || slice.err <= 0)
+		luaL_error(L, "slice must be greater than 0");
+	struct fiber_slice current_slice = cord()->current_slice;
+	slice.warn += current_slice.warn;
+	slice.err += current_slice.err;
+	fiber_set_current_slice(slice);
```

Comment:
It looks ugly that you have a helper that assigns `cord()->current_slice`, but at the same time you read `cord()->current_slice` directly. Better drop the helper.

Reply:
fixed